### ISSUE TITLE
Plugin problems with Ryzom Core studio

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#186]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/186) by yann k (Bitbucket: [yannk2](https://bitbucket.org/yannk2), GitHub: [yannk2](https://github.com/yannk2)).**

----------------------------------------

I use RC studio on a Windows XP VM on my Linux computer.  I have installed a version of RC Studio compiled for me based upon. It is a Ryzom Core Studio version compiled on Aug 30 2014 14:30:27.

When I launch studio.exe, I still have the problem of plugins who don't want to load :
"studio_plugin_georges_editor_r.dll : Cannot load library
E:/RC_studio/RyzomCore/plugins/studio_plugin_georges_editor_r.dll: The specified module could not be found.

studio_plugin_gui_editor_r.dll : Cannot load library
E:/RC_studio/RyzomCore/plugins/studio_plugin_gui_editor_r.dll: The specified module could not be found.

studio_plugin_landscape_editor_r.dll : Cannot load library
E:/RC_studio/RyzomCore/plugins/studio_plugin_landscape_editor_r.dll: The specified module could not be found.

studio_plugin_tile_editor_r.dll : File does not exist:
E:/RC_studio/RyzomCore/plugins/studio_plugin_tile_editor_r.dll

studio_plugin_translation_manager_r.dll : Cannot load library
E:/RC_studio/RyzomCore/plugins/studio_plugin_translation_manager_r.dll: The specified module could not be found.

studio_plugin_world_editor_r.dll : Cannot load plugin because dependency failed to load: Landscape editor"

The file "E:/RC_studio/RyzomCore/plugins/studio_plugin_tile_editor_r.dll" is not actually there, that's right, I don't know why but the other ones are there, it just can't load them.
WHen I try to manually load them after, with the new tool, it says it can't load them either.

**Comment by ryzom-pipeline** [comment]
**Original comment by yann k (Bitbucket: [yannk2](https://bitbucket.org/yannk2), GitHub: [yannk2](https://github.com/yannk2)).**

----------------------------------------

Here is my log.log file

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

Had the same issue during the summer, with debug builds.
I'm just going to write this here as a note:
Had to rebuild Qt for it to load the plugins.

( Obviously that's not the solution for this problem, will have to do further research to find out what's wrong )

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

I would have fixed it already if I had a solution.
Either way you can try the workaround I posted...

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Is this the issue with conflicting zlib libraries? If so, it's the fault of the one who built it, not a bug. Need to use the same zlib dll for Qt build as for our code. Can't pull zlib in statically.

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

No it's not related to zlib.
The plugins just can't be loaded.

**Comment by ryzom-pipeline** [comment]
**Original comment by chris wade (Bitbucket: [crehop2](https://bitbucket.org/crehop2), ).**

----------------------------------------

Sorry if I'm a complete noob laszlo, I didn't understand your solution  :( "Had to rebuild Qt for it to load the plugins" as I had downloaded and installed QT and didn't build it. Can you be a bit more detailed?

**Comment by ryzom-pipeline** [comment]
**Original comment by chris wade (Bitbucket: [crehop2](https://bitbucket.org/crehop2), ).**

----------------------------------------

Or maybe provide a precompiled studio?, as it took me a long time to actually get it compiled without errors since the dependencies and some of the dependencies dependencies require compiling also, would save a lot of people a lot of time and help the project I think.

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

I don't really know what you are talking about. You only have to build Ryzom, not the dependencies.
The dependencies are provided in the externals package ( for Windows that is ).
As for binary packages of the studio, that's a good idea, but who's going to do this, and why just the studio and not everything that is in the repo? :)

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

Oh yea and also, yes I downloaded Qt as well, but that package contains the source code and build files as well, so you can rebuild it if you want :)

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Yes, you must build Qt manually, and specify to use the zlib from our externals package.

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

Jan Boon, again that's a different bug :D

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Nah, I'm sure that's his issue. 100% if he's using stock Qt.

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

With the zlib issue you can't even build the core and tile editor plugins, here they just cannot be loaded.
As in the files exist, but yet they can't be loaded.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Ok

**Comment by ryzom-pipeline** [comment]
**Original comment by chris wade (Bitbucket: [crehop2](https://bitbucket.org/crehop2), ).**

----------------------------------------

Well Laszlo he may be correct I'm kinda a newb... I Pointed the zlib to Qcore4.lib which compiled it without errors as using the regular Zlib throws already defined in Qcore4.lib errors. Could that be the cause?

**Comment by ryzom-pipeline** [comment]
**Original comment by chris wade (Bitbucket: [crehop2](https://bitbucket.org/crehop2), ).**

----------------------------------------

When I try to compile with the provided externals this is what I get :( Im really trying to do this without asking questions but I have been trying for 2 weeks and still cannot compile it, I have followed all the instructions on the website and it give me this. How to I fix this?

D:\ryzom\CoreStudio\ryzom\tools\client\client_config_qt\MSVCRT.lib(crtexew.obj)	Ryzom, Tools - Ryzom Configuration Qt
Error	9	error LNK2005: _inflateReset already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\core\zlib.lib(inflate.obj)	Tools - Studio Plugin - Core
Error	18	error LNK2005: _inflateReset already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\example\zlib.lib(inflate.obj)	Tools - Studio Plugin - Example
Error	23	error LNK2005: _inflateReset already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\tile_editor\zlib.lib(inflate.obj)	Tools - Studio Plugin - Tile Editor
Error	10	error LNK2005: _inflateInit2_ already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\core\zlib.lib(inflate.obj)	Tools - Studio Plugin - Core
Error	19	error LNK2005: _inflateInit2_ already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\example\zlib.lib(inflate.obj)	Tools - Studio Plugin - Example
Error	24	error LNK2005: _inflateInit2_ already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\tile_editor\zlib.lib(inflate.obj)	Tools - Studio Plugin - Tile Editor
Error	12	error LNK2005: _inflateEnd already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\core\zlib.lib(inflate.obj)	Tools - Studio Plugin - Core
Error	21	error LNK2005: _inflateEnd already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\example\zlib.lib(inflate.obj)	Tools - Studio Plugin - Example
Error	26	error LNK2005: _inflateEnd already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\tile_editor\zlib.lib(inflate.obj)	Tools - Studio Plugin - Tile Editor
Error	11	error LNK2005: _inflate already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\core\zlib.lib(inflate.obj)	Tools - Studio Plugin - Core
Error	20	error LNK2005: _inflate already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\example\zlib.lib(inflate.obj)	Tools - Studio Plugin - Example
Error	25	error LNK2005: _inflate already defined in QtCore4.lib(QtCore4.dll)	D:\ryzom\CoreStudio\studio\src\plugins\tile_editor\zlib.lib(inflate.obj)	Tools - Studio Plugin - Tile Editor
Error	13	error LNK1169: one or more multiply defined symbols found	D:\ryzom\CoreStudio\bin\Release\studio_plugin_core_r.dll	Tools - Studio Plugin - Core
Error	22	error LNK1169: one or more multiply defined symbols found	D:\ryzom\CoreStudio\bin\Release\studio_plugin_example_r.dll	Tools - Studio Plugin - Example
Error	27	error LNK1169: one or more multiply defined symbols found	D:\ryzom\CoreStudio\bin\Release\studio_plugin_tile_editor_r.dll	Tools - Studio Plugin - Tile Editor
Error	15	error LNK1120: 1 unresolved externals	D:\ryzom\CoreStudio\bin\Release\ryzom_configuration_qt_r.exe	Ryzom, Tools - Ryzom Configuration Qt

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

That's another issue, which has nothing to do with this one.
That is indeed the one Jan Boon mentioned.
But again that has nothing to do with #186

**Comment by ryzom-pipeline** [comment]
**Original comment by chris wade (Bitbucket: [crehop2](https://bitbucket.org/crehop2), ).**

----------------------------------------

Ok, tried everything Re-compiled QT with -system-zlib option which did fix the compilation errors. Uninstalled everything started fresh and recompiled everything with no error's, however I am still getting the same issue 
![Untitled.jpg](https://bitbucket.org/repo/R9LxXq/images/3560513603-Untitled.jpg) also I was playing around in the editor in its current state and also got this error if it helps any A failed assertion occurs
ProcName: studio.exe
Date: 2014/10/17 16:29:08
File: ..\..\..\..\ryzomCode\code\nel\src\misc\i_xml.cpp
Line: 196
FuncName: NLMISC::CIXml::init
Reason: "seekGood" 
-------------------------------

No log
-------------------------------

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Please run Dependency Walker on the dll's that cannot load, and make sure the dependencies can be found by the application.

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

Here's a package as requested:
http://www.mediafire.com/download/ne6yi2j17hhgyi9/studio.zip

**Comment by ryzom-pipeline** [comment]
**Original comment by dfighter1985 (Bitbucket: [dfighter1985](https://bitbucket.org/dfighter1985), GitHub: [dfighter1985](https://github.com/dfighter1985)).**

----------------------------------------

OK, I think this can be closed.

For future reference, I've checked what I needed to run the plugins exactly, when built with Visual C++ 2010 using Qt 4.8.5:

* Qtcore4.dll

* QtGui4.dll

* QtNetwork4.dll

* QtOpenGL4.dll

* QtXml4.dll

* libeay32.dll

* ssleay32.dll

The Qt dlls can be found in the Qt packages, the libeay32 and ssleay32 files come from OpenSSL.

Also note this was not in any way a bug.
